libflux/flux-core/src/formatter/mod.rs

```diff
@@ -475,8 +475,42 @@ impl Formatter {
 
     fn format_function_expression(&mut self, n: &ast::FunctionExpr) {
         self.format_comments(&n.lparen);
-        let multiline = n.params.len() > 4 && n.base.is_multiline();
+        let mut multiline = false;
         self.write_rune('(');
+
+        let mut temp = Formatter {
+            builder: String::new(),
+            indentation: self.indentation.clone(),
+            clear: self.clear.clone(),
+            temp_indent: self.temp_indent.clone(),
+            err: self.err.clone(),
+            temp_singleline: self.temp_singleline.clone(),
+        };
+
+        let mut sep;
+
+        for (i, property) in (&n.params).iter().enumerate() {
+            if let Some(v) = &property.value{
+                temp.format_node(&Node::from_expr(v));
+            };
+
+            // if any child node adds newlines, then parent node will be multiline
+            if temp.builder.contains("\n"){
+                multiline = true;
+            }
+
+            if i != 0 {
+                sep = ",\n";
+                temp.write_string(sep);
+                if multiline {
+                    temp.write_indent();
```

Comment:
If only the second, or third etc argument is spread over multiple lines the the first won't get indented I think. Like

```
import "array"
array.from(foo: false,
    rows: [
        {_value: "one", _time: 2021-08-30T00:00:00Z},
        {_value: "one", _time: 2021-08-30T00:00:00Z},
        {_value: "one", _time: 2021-08-30T00:00:00Z},
        {_value: "one", _time: 2021-08-30T00:00:00Z},
        {_value: "one", _time: 2021-08-30T00:00:00Z},
        {_value: "one", _time: 2021-08-30T00:00:00Z},
        {_value: "one", _time: 2021-08-30T00:00:00Z},
    ],
)"
```